meta-balena-common/recipes-bsp/grub/grub-efi_%.bbappend

```diff
@@ -9,5 +35,40 @@ do_deploy:append:class-target() {
     install -d ${DEPLOYDIR}/grub/${GRUB_TARGET}-efi/
 }
 
-# build in additional required modules
-GRUB_BUILDIN:append = " regexp probe gzio"
+do_mkimage() {
+    cd ${B}
+
+    if [ "${SIGN}" = "true" ]; then
+        GRUB_PUBKEY_ARG="--pubkey "${B}/pubkey.gpg""
+    fi
+
+    # Search for the grub.cfg on the local boot media by using the
+    # built in cfg file provided via this recipe
+    grub-mkimage -c ../cfg -p ${EFIDIR} -d ./grub-core/ \
+           -O ${GRUB_TARGET}-efi -o ./${GRUB_IMAGE_PREFIX}${GRUB_IMAGE} \
+           ${GRUB_BUILDIN} ${GRUB_PUBKEY_ARG}
+}
+
+do_sign_efi() {
+    if [ "${SIGN}" != "true" ]; then
+        return 0
+    fi
+
+    export CURL_CA_BUNDLE="${STAGING_DIR_NATIVE}/etc/ssl/certs/ca-certificates.crt"
+
+    EFI_APP="${B}/${GRUB_IMAGE_PREFIX}${GRUB_IMAGE}"
+    REQUEST_FILE=$(mktemp)
+    RESPONSE_FILE=$(mktemp)
+    echo "{\"key_id\": \"${SIGN_EFI_KEY_ID}\", \"payload\": \"$(base64 -w 0 ${EFI_APP})\"}" > "${REQUEST_FILE}"
+    curl --fail "${SIGN_API}/secureboot/efi" -X POST -H "Content-Type: application/json" -H "X-API-Key: ${SIGN_API_KEY}" -d "@${REQUEST_FILE}" > "${RESPONSE_FILE}"
+    jq -r .signed < "${RESPONSE_FILE}" | base64 -d > "${EFI_APP}.signed"
+    rm -f "${REQUEST_FILE}" "${RESPONSE_FILE}"
+
+    mv "${EFI_APP}.signed" "${EFI_APP}"
+}
+
+do_sign_efi:class-native() {
+    :
+}
```

Comment:
Investigate why grub-efi builds native

Reply:
grub-efi has an explicit dependency on `grub-native` that bring this in.
https://git.yoctoproject.org/cgit/cgit.cgi/poky/tree/meta/recipes-bsp/grub/grub-efi_2.04.bb?h=dunfell#n7